Stratis.Bitcoin.Features.Consensus/ConsensusLoop.cs

```diff
@@ -78,41 +90,56 @@ public BlockResult ExecuteNextBlock(CancellationToken cancellationToken)
                     {
                         result.Block = this.Puller.NextBlock(cancellationToken);
                         if (result.Block != null)
+                        {
+                            this.logger.LogTrace("Block received from puller.");
                             break;
-                        else
+                        }
+
+                        this.logger.LogTrace("No block received from puller due to reorganization, rewinding.");
+                        while (true)
                         {
-                            while (true)
+                            uint256 hash = this.UTXOSet.Rewind().GetAwaiter().GetResult();
+                            ChainedBlock rewinded = this.Chain.GetBlock(hash);
+                            if (rewinded == null)
                             {
-                                var hash = this.UTXOSet.Rewind().GetAwaiter().GetResult();
-                                var rewinded = this.Chain.GetBlock(hash);
-                                if (rewinded == null)
-                                    continue;
-                                this.Tip = rewinded;
-                                this.Puller.SetLocation(rewinded);
-                                break;
+                                this.logger.LogTrace("Rewound to '{0}', which is still not a part of the current best chain, rewinding further.", hash);
```

Comment:
`Rewound` no?

Reply:
as for what i found, rewound should be past tense of rewind, so this should be correct, but no native speaker here :)